lms/djangoapps/grades/tests/integration/test_events.py

```diff
@@ -0,0 +1,126 @@
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+from courseware.models import StudentModule
+from courseware.tests.test_submitting_problems import ProblemSubmissionTestMixin
+from student.models import CourseEnrollment
+from student.tests.factories import UserFactory
+from ...new.subsection_grade import SubsectionGradeFactory
+
+
+class DeleteStateEventIntegrationTest(ProblemSubmissionTestMixin, SharedModuleStoreTestCase):
+    """
+    Tests integration between the eventing in various layers
+    of the grading infrastructure.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(DeleteStateEventIntegrationTest, cls).setUpClass()
+        cls.course = CourseFactory.create()
+        cls.chapter = ItemFactory.create(
+            parent=cls.course,
+            category="chapter",
+            display_name="Test Chapter"
+        )
+        cls.sequence = ItemFactory.create(
+            parent=cls.chapter,
+            category='sequential',
+            display_name="Test Sequential 1",
+            graded=True,
+            format="Homework"
+        )
+        cls.vertical = ItemFactory.create(
+            parent=cls.sequence,
+            category='vertical',
+            display_name='Test Vertical 1'
+        )
+        problem_xml = MultipleChoiceResponseXMLFactory().build_xml(
+            question_text='The correct answer is Choice 3',
+            choices=[False, False, True, False],
+            choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
+        )
+        cls.problem = ItemFactory.create(
+            parent=cls.vertical,
+            category="problem",
+            display_name="Test Problem",
+            data=problem_xml
+        )
+
+    def setUp(self):
+        super(DeleteStateEventIntegrationTest, self).setUp()
+        self.request = get_mock_request(UserFactory())
+        self.student = self.request.user
+        self.client.login(username=self.request.user.username, password="test")
+        CourseEnrollment.enroll(self.request.user, self.course.id)
+        course_structure = get_course_blocks(self.request.user, self.course.location)
+        self.subsection_factory = SubsectionGradeFactory(
+            self.request.user,
+            course_structure=course_structure,
+            course=self.course,
+        )
+        self.instructor = UserFactory.create(is_staff=True, username=u'test_instructor', password=u'test')
+
+    @patch('lms.djangoapps.instructor.enrollment.tracker')
+    @patch('lms.djangoapps.grades.signals.handlers.tracker')
+    @patch('lms.djangoapps.grades.models.tracker')
+    def test_delete_student_state_events(self, models_tracker, handlers_tracker, enrollment_tracker):
+        # submit answer
+        StudentModule.objects.create(
+            student=self.student,
+            course_id=self.course.id,
+            module_state_key=self.problem.location,
+            state=json.dumps({'attempts': 10}),
+        )
```

Comment:
Why is this line needed?  Doesn't the call to `submit_question_answer` on the following line create the `StudentModule` entry?